ci-operator/step-registry/telco5g/cluster-setup/telco5g-cluster-setup-commands.sh

```diff
@@ -232,9 +239,9 @@ cat << EOF > ~/fetch-information.yml
     shell: kcli ssh root@${CLUSTER_NAME}-installer 'oc get node'
 EOF
 
-status=0
-ANSIBLE_STDOUT_CALLBACK=debug ansible-playbook -i $SHARED_DIR/inventory ~/ocp-install.yml -vv || status=$?
-ansible-playbook -i $SHARED_DIR/inventory ~/fetch-kubeconfig.yml -vv || true
+if [[ "$T5CI_JOB_TYPE" != "cnftests" ]]; then
```

Comment:
We need to keep here possibility for other jobs to fetch kubeconfig if exists even if cluster failed, for next `gather-pao` step which will gather debug information. That's why I put here `|| true` after each call. This should remain the same.
For case of cnf-tests job it shouldn't be the case, if cluster is not installed properly, just fail .
Please rethink the logic here including all above.

Reply:
added a variable `$NOT_CNFTESTS` that is set to 'false' unless the job type is `cnftests`

so in normal execution, it will still evaluate to `|| true`, which will allow the gather-pao step to work. however, during CNF tests, it will evaluate to `|| false`, which accomplishes the desired outcome:

```
[gpuckett@fedora release]$ `/bin/false` || `/bin/false`
[gpuckett@fedora release]$ echo $?
1
[gpuckett@fedora release]$ `/bin/true` || `/bin/false`
[gpuckett@fedora release]$ echo $?
0
[gpuckett@fedora release]$ `/bin/false`
[gpuckett@fedora release]$ echo $?
1
```

Reply:
In case of `cnftests` job run and cluster installation failed, we don't want to run any on these playbooks. But we need to ensure cluster is functional or not. I think about adding a specific playbook to check it. Or task in any of existing playbooks, anyway cluster should be up for any job to run.

cluster is up:
```
$ kcli ssh root@cnfdu9-installer "oc get clusterversion"
NAME      VERSION                              AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.12.0-0.nightly-2023-01-24-150415   True        False         20h     Cluster version is 4.12.0-0.nightly-2023-01-24-150415
```
cluster is not up:
```
[kni@cnfdu0 ~]$ kcli ssh root@cnfdu3-installer "oc get clusterversion"
NAME      VERSION   AVAILABLE   PROGRESSING   SINCE   STATUS
version             False       True          31h     Unable to apply 4.11.0-0.nightly-2023-01-25-044931: the cluster operator monitoring has not yet successfully rolled out
```